### PR TITLE
Filings dashboard - Legal Obligation

### PR DESCRIPTION
**Pull request by lekshmimallika-aot**
*Issue #:* /bcgov/entity#3828

*Description of changes:*
Added legal obligation section in filing dashboard
Added tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by severinbeauvais** [comment]
GHA unit tests failed. You may have a conflict with something Cam did recently.

File: src/components/Dashboard/LegalObligation.vue [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  <div v-if="showLegalObligation && isBusinessWithNoMaintenanceFilings" class="legal-obligation-container">
+    <v-card flat class="legal-obligation-section">
+      <v-icon color="blue darken-2" class="info-icon">mdi-information-outline</v-icon>
+      <div class="share-structure-check-text">
+        <span class="bold-text">Legal Obligations:</span> You are required by the
+        <a href="http://www.bclaws.ca/civix/document/id/complete/statreg/02057_02" target="_blank">
+          Business Corporations Act
+        </a>
+        to keep the information about your corporation up to date with the Registrar: For example, you must file annual
+        reports,director changes and address changes.
```
Comment (severinbeauvais) [inline src/components/Dashboard/LegalObligation.vue:11]: Add a space after comma.

File: src/components/Dashboard/LegalObligation.vue [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                (appointing, ceasing or updating an existing director's name and address)
+              </li>
+              <li>
+                <span class="bold-text">Company address changes</span> - update any changes to a company's registered or
+                records office addresses as they occur.
+              </li>
+            </ul>
+          </div>
+          <div class="read-more-line">
+            <span
```
Comment (severinbeauvais) [inline src/components/Dashboard/LegalObligation.vue:37]: Fix weird line break.

File: src/components/Dashboard/LegalObligation.vue [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    ...mapGetters(['isBusinessWithNoMaintenanceFilings'])
+  }
+})
+export default class LegalObligation extends Vue {
+  readonly isBusinessWithNoMaintenanceFilings!: boolean
+  private readMoreFlag: boolean = false
+  private showLegalObligation: boolean = true
+
+  private get detailInfoURL () {
+    return 'https://www2.gov.bc.ca/gov/content/employment-business/business/' +
```
Comment (severinbeauvais) [inline src/components/Dashboard/LegalObligation.vue:71]: OMG. Good call on using a getter :)

File: src/store/getters.ts [hunk 4]
```diff
@@ -77,5 +77,14 @@ export default {
       }
     }
     return lastFilingDate
+  },
+
+  isBusinessWithNoMaintenanceFilings (state): boolean {
+    if (state.filings && state.filings.length === 1 &&
```
Comment (severinbeauvais) [inline src/store/getters.ts:83]: Please add a comment explaining why this code is needed / how it works. I guess, something like "this is to show the Legal Obligations only for a new business that hasn't filing anything else".

BTW, if they have an Annual Report in their Todo List, does this display? Seems a bit weird but I dunno.

**Comment by cameron-freshworks** [comment]
> GHA unit tests failed. You may have a conflict with something Cam did recently.

@severinbeauvais My last commit to this repo was June 2 updating the common components. Not sure how that would affect anything at this point. My other pr is in code review.

File: tests/unit/LegalObligation.spec.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+      },
+      'annualReport': {
+        'annualGeneralMeetingDate': '2019-12-31',
+        'annualReportDate': '2019-12-31'
+      }
+    }
+  }
+]
+
+async function waitForUpdate () {
```
Comment (cameron-freshworks) [inline tests/unit/LegalObligation.spec.ts:72]: I always enjoy your custom methods inside the unit tests :)

File: src/components/Dashboard/LegalObligation.vue (hunk 3 above)
Comment (severinbeauvais) [reply src/components/Dashboard/LegalObligation.vue:71]: @cameron-freshworks Did you see the length of this URL?

Comment (cameron-freshworks) [reply src/components/Dashboard/LegalObligation.vue:71]: 😆 Yea, i noticed after! I didn't understand at first hahah

File: src/store/getters.ts (hunk 4 above)
Comment (lekshmimallika-aot) [reply src/store/getters.ts:83]: Sure will do

File: tests/unit/LegalObligation.spec.ts (hunk 5 above)
Comment (lekshmimallika-aot) [reply tests/unit/LegalObligation.spec.ts:72]: Thanks :)

**Comment by lekshmimallika-aot** [comment]
> Should this be a dev dependency?
> Also, please commit the package lock file changes (after `npm i`).

That dint work. I am tryingdifferent options

**Comment by severinbeauvais** [comment]
> > Should this be a dev dependency?
> > Also, please commit the package lock file changes (after `npm i`).
> 
> That dint work. I am tryingdifferent options

Looks like you got it :)

File: src/store/getters.ts (hunk 4 above)
Comment (lekshmimallika-aot) [reply src/store/getters.ts:84]: Ya, It doesnt make sense to show this if there is something in the todo list. I will add that condition

**Pull request merged**